src/drivers/srf02_i2c/srf02_i2c.cpp

```diff
@@ -733,21 +760,25 @@ start(uint8_t rotation)
 	g_dev = new SRF02_I2C(rotation, SRF02_I2C_BUS);
 
 	if (g_dev == nullptr) {
+		err(1,"SRF02_I2C constructor failed");
```

Comment:
Can you change these to PX4_ERR, followed by return? This err actually exits, which is fine on NuttX, but when running on Linux exits the entire running firmware.